src/python/pants/option/options_bootstrapper.py

```diff
@@ -3,255 +3,144 @@
-            path_list_values.append(ListValueComponent.create(default))
-        for var in evars:
-            if var in env:
-                path_list_values.append(ListValueComponent.create(env[var]))
-                break
-
-        for arg in args:
-            # Technically this is very slightly incorrect, as we don't check scope.  But it's
-            # very unlikely that any task or subsystem will have an option named --pants-config-files.
-            # TODO: Enforce a ban on options with a --pants- prefix outside our global options?
-            if arg.startswith(flag):
-                path_list_values.append(ListValueComponent.create(arg[len(flag) :]))
-
-        return ListValueComponent.merge(path_list_values).val
-
-    @staticmethod
-    def parse_bootstrap_options(
-        env: Mapping[str, str], args: Sequence[str], config: Config
-    ) -> Options:
-        bootstrap_options = Options.create(
-            env=env,
-            config=config,
-            known_scope_infos=[GlobalOptions.get_scope_info()],
-            args=args,
-            native_options_config_discovery=False,
-        )
-
-        for options_info in collect_options_info(BootstrapOptions):
-            # Only use of Options.register?
-            bootstrap_options.register(GLOBAL_SCOPE, *options_info.args, **options_info.kwargs)
-
-        return bootstrap_options
+        return f"OptionsBootstrapper(args={args}, env={env})"
 
     @classmethod
     def create(
-        cls, env: Mapping[str, str], args: Sequence[str], *, allow_pantsrc: bool
+        cls,
+        args: Sequence[str],
+        env: Mapping[str, str],
+        *,
+        allow_pantsrc: bool = True,
     ) -> OptionsBootstrapper:
         """Parses the minimum amount of configuration necessary to create an OptionsBootstrapper.
 
-        :param env: An environment dictionary.
         :param args: An args array.
+        :param env: An environment dictionary.
         :param allow_pantsrc: True to allow pantsrc files to be used. Unless tests are expecting to
           consume pantsrc files, they should pass False in order to avoid reading files from
           absolute paths. Production use-cases should pass True to allow options values to make the
           decision of whether to respect pantsrc files.
         """
         args = tuple(args)
-        with warnings.catch_warnings(record=True):
-            # We can't use pants.engine.fs.FileContent here because it would cause a circular dep.
-            @dataclass(frozen=True)
-            class FileContent:
-                path: str
-                content: bytes
-
-            def filecontent_for(path: str) -> FileContent:
-                return FileContent(
-                    ensure_text(path),
-                    read_file(path, binary_mode=True),
-                )
-
-            bargs = cls._get_bootstrap_args(args)
-
-            config_file_paths = cls.get_config_file_paths(env=env, args=args)
```

Comment:
This whole bootstrapping sequence is now implemented in Rust, and is completely unnecessary in Python.